core/shared/src/main/scala/org/threeten/bp/Duration.scala

```diff
@@ -953,9 +955,37 @@ final class Duration private (private val seconds: Long, private val nanos: Int)
    * @return
    *   the total length of the duration in seconds, with a scale of 9, not null
    */
-  private def toSeconds: BigDecimal =
+  private def toSecondsBD: BigDecimal =
     BigDecimal.valueOf(seconds).add(BigDecimal.valueOf(nanos.toLong, 9))
 
+  /**
+   * Gets the number of seconds in this duration.
+   *
+   * This returns the total number of whole seconds in the duration.
+   *
+   * This instance is immutable and unaffected by this method call.
+   *
+   * @return
+   *   the whole seconds part of the length of the duration, positive or negative
+   *
+   * @since 9
+   */
+  def toSeconds: Long = seconds
+
+  /**
+   * Extracts the number of seconds part in the duration.
+   *
+   * This returns the remaining seconds when dividing toSeconds() by seconds in a minute. This is
+   * based on the standard definition of a minute as 60 seconds.
+   *
+   * This instance is immutable and unaffected by this method call.
+   *
+   * @return
+   *   the number of seconds parts in the duration, may be negative
+   * @since 9
+   */
+  def toSecondsPart: Int = (toSeconds / SECONDS_PER_MINUTE).toInt
```

Comment:
Hmm, shouldn't this be using `%` to get the remainder instead of `/` ? Why doesn't the test suite catch this?

Reply:
Oh right, it say so on the docs :smile: the test suite is not good enough

Reply:
Great catch I added a fix